diff --git a/node_modules/whatsapp-web.js/src/Client.js b/node_modules/whatsapp-web.js/src/Client.js
index 800109e..5d7dcb3 100644
--- a/node_modules/whatsapp-web.js/src/Client.js
+++ b/node_modules/whatsapp-web.js/src/Client.js
@@ -119,21 +119,10 @@ class Client extends EventEmitter {
         ) {
             this.options.authTimeoutMs = 30000;
         }
-        let start = Date.now();
-        let timeout = this.options.authTimeoutMs;
-        let res = false;
-        while (start > Date.now() - timeout) {
-            res = await this.pupPage.evaluate(
-                'window.Debug?.VERSION != undefined',
-            );
-            if (res) {
-                break;
-            }
-            await new Promise((r) => setTimeout(r, 200));
-        }
-        if (!res) {
-            throw 'auth timeout';
-        }
+        await this.pupPage.waitForFunction(
+            'window.Debug?.VERSION != undefined',
+            { timeout: this.options.authTimeoutMs },
+        );
         await this.setDeviceName(
             this.options.deviceName,
             this.options.browserName,
